src/mon/FSCommands.cc

```diff
@@ -1541,6 +1514,13 @@ int FileSystemCommandHandler::_check_pool(
 
   const string& pool_name = osd_map.get_pool_name(pool_id);
 
+  if (!allow_overlay && pool->application_metadata.count(pg_pool_t::APPLICATION_NAME_CEPHFS) != 0) {
+    auto kv_pair = pool->application_metadata.find(pg_pool_t::APPLICATION_NAME_CEPHFS)->second.begin();
+    *ss << "RADOS pool '" << pool_name << "' is already used by filesystem '" << kv_pair->second << "'";
+    return -EINVAL;
+  }
```

Comment:
Why not do this in https://github.com/ceph/ceph/pull/44900/commits/a00daf381445fa8f542f8682fa4caee7cd2b7c6a#diff-42ceb20fdeb082bdba8efcfa88a042d19932504d90a8c13c8cc0392daf9e12b3R1568-R1574 ?

Reply:
Done. Squashed commits.